### PR TITLE
Fix HTML entities not decoded in comment just added

### PR DESCRIPTION
**Pull request by danxuliu**
Follow up to #29029

The XML data received from the comments endpoint [has an inconsistent encoding](https://github.com/nextcloud/server/issues/29306); some entities are encoded once and others are encoded twice. When the comment list is loaded the comments are fetched using `GetComments`, [which handles all that](https://github.com/nextcloud/server/blob/9d26671f05275055619960a66981201ed660bdd2/apps/comments/src/services/GetComments.js#L84-L87), and therefore shows the messages and author names as expected.

However, when a new comment is posted the list is not got again; instead the new comment [is loaded](https://github.com/nextcloud/server/blob/cba636124a1574521bf6aadd28a47ed3049e74bf/apps/comments/src/views/Comments.vue#L257-L264) from [the comment data returned after posting it](https://github.com/nextcloud/server/blob/14e57bf70a612c16eccdd733b3562d46053419c5/apps/comments/src/mixins/CommentMixin.js#L102-L104). This is done in `NewComment`, which did not decode the messages nor the author names, and therefore showed, for example, `&amp;` instead of `&`.

To solve that now the same decoding logic used in `GetComments` is applied too in `NewComment`.

@Pytal I extracted _decodeHtmlEntities_ to its own file and I assigned the copyright of that file to you, as [you added the original function](https://github.com/nextcloud/server/pull/29029/files#diff-da66023dddd7ab71ef54bf814084c988c8f12c4bb14c4f1e269734058dd5e042R89-R96). I hope that is fine :-)

## How to test

- Set the display name of the user to `John "The Hangman" Ruth`
- Open the Files app
- Show the sidebar for any file
- Open the _Comments_ tab
- Send a new comment with text `< & >`

### Result with this pull request

The text of the comment just added is `< & >`, and it was sent by `John "The Hangman" Ruth`

### Result without this pull request

The text of the comment just added is `&amp;lt; &amp; &amp;gt;`, and it was sent by `John &quot;The Hangman&quot; Ruth`; if the page is reloaded and the _Comments_ tab is shown again now the message has the expected author and text

### CODE REVIEW

**Review by violethaze74 — APPROVED** [review]

File: apps/comments/src/utils/decodeHtmlEntities.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * @param {any} value -
+ * @param {any} passes -
+ */
+export default function decodeHtmlEntities(value, passes = 1) {
```
Comment (Pytal) [inline apps/comments/src/utils/decodeHtmlEntities.js:27]:
```suggestion
export function decodeHtmlEntities(value, passes = 1) {
```
Let's go with a named export for better IDE integration

**Review by Pytal — APPROVED** [review]
Other than the named export suggestion :)

**Comment by danxuliu** [comment]
> Other than the named export suggestion :)

Done :-)

**Review by szaimen — APPROVED** [review]
🐘

**Pull request merged**